### PR TITLE
Begin migrating `upload_analyzer` scripts to graplctl

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/291

### What changes does this PR make to Grapl? Why?
- Provides an "upload analyzers" functionality through `graplctl upload analyzer`
- Refactors a lot of the existing upload-analyzers code into its own file

### How were these changes tested?
I wrote test cases :')

### CODE REVIEW

File: src/python/grapl-tests-common/grapl_tests_common/upload_analyzers.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, NamedTuple, Sequence
+
```
Comment (wimax-grapl) [inline src/python/grapl-tests-common/grapl_tests_common/upload_analyzers.py:5]: # This file is just types/functions refactored out of other files.

**Review by wimax-grapl — COMMENTED** [review]
left comment

File: src/python/graplctl/tests/test_graplctl.py [hunk 2]
```diff
@@ -1,59 +1,5 @@
-from contextlib import contextmanager
```
Comment (wimax-grapl) [inline src/python/graplctl/tests/test_graplctl.py:1]: this file was split up into
shared
test_thing
test_other_thing

File: src/python/graplctl/tests/fake_uploads/BUILD [hunk 3]
```diff
@@ -0,0 +1 @@
+python_library()
```
Comment (jgrillo-grapl) [inline src/python/graplctl/tests/fake_uploads/BUILD:1]: Why does each of these subpackages get its own BUILD file?

**Review by jgrillo-grapl — APPROVED** [review]
LGTM overall, just one question

File: src/python/graplctl/tests/fake_uploads/BUILD (hunk 3 above)
Comment (jgrillo-grapl) [reply src/python/graplctl/tests/fake_uploads/BUILD:1]: Answered offline

**Pull request merged**